linode/manager
# upcoming: [M3-8378] - OBJ Cleanup

## Related
- [x] https://github.com/linode/manager/pull/10797

## Changes  🔄
- Removed our react hook form logic for `BucketRateLimitTable` and corresponding files. This is far from being ready and it introduced complexity since this component is used in 3 different places. This will most likely be a 2025 initiative to add this logic in when it's ready.
    - By removing this, we fixed an issue where app was crashing when trying to display the bucket rate table in any of the drawers for E2/E3. 
- Moved the "Bucket Rate Limits" title and corresponding helper text into `BucketRateLimitTable.tsx` since we were repeating that in 3 different places.
- The Create Bucket drawer now displays each additional step once we have the preceding information selected.
   - Once we select a region, we'll show endpoint types
   - Once we select endpoint types, we'll show bucket rate limits (except for E0/E1 of course) 

## Target release date 🗓️
N/A

## How to test 🧪

### Prerequisites
- Enable multi-cluster and gen2 flags
- Enable MSW

### Reproduction steps
(How to reproduce the issue, if applicable)
- When selecting a E2/E3 endpoint type bucket, app would crash because we didn't implement our form logic to `BucketDetailsDrawer`

### Verification steps
(How to verify changes)
- Pull down branch or preview link

- Select "Details" for any E2/E3 endpoint bucket:
   - Drawer should open and bucket rate limits table should show

- Create a bucket and observe:
   - Once a region is selected, endpoint types show
   - Once we select endpoint types, bucket rate limits show (except for E0/E1 of course) 

- Go to bucket details page and then properties tab
   - Ensure title, description, and table all look good
   - Do this for E0 -> E3 endpoint types

## As an Author I have considered 🤔

*Check all that apply*
